### PR TITLE
[RFC] Introduce doctest

### PR DESCRIPTION
**Pull request by hvy**
Rather minimal change to introduce doctest without breaking CI. Code snippets (e.g. `Example:`) should be updated to run in doctest and fixed as necessary. 

You can easily run doctest locally
1. `cd docs`.
2. `make doctest`.

### CODE REVIEW

File: optuna/study.py [hunk 1]
```diff
@@ -306,12 +306,27 @@ def trials_dataframe(self, include_internal_fields=False):
...
-            >>> df.params.x[0]
-            1.0
+            .. testsetup:: *
+
+                import optuna
+
+                def objective(trial):
+                    trial.report(trial.number)
+                    trial.suggest_uniform('x', float(trial.number), float(trial.number))
+                    return trial.number
```
Comment (g-votte) [inline optuna/study.py]: We may want to change the scenario. Returning a trial number is somewhat unrealistic as a user scenario.

File: optuna/study.py [hunk 2]
```diff
@@ -306,12 +306,27 @@ def trials_dataframe(self, include_internal_fields=False):
...
-            >>> df
-            >>> df.value[0]
-            0.0
-            >>> df.params.x[0]
-            1.0
+            .. testsetup:: *
+
+                import optuna
+
+                def objective(trial):
```
Comment (g-votte) [inline optuna/study.py]: How about moving `objective` to the  `doctest` block? I prefer an example's being self-contained unless it's too noisy (e.g., too many imports).

Comment (hvy) [reply optuna/study.py]: Moved everything to a single `.. testcode::` (which in contrast to `.. doctest::` does check the outputs) with a more realistic scenario. It became a lot more comprehensible not hiding parts of the code as you suggested.

File: optuna/study.py (hunk 1 above)
Comment (hvy) [reply optuna/study.py]: Addressed this comment as well in https://github.com/optuna/optuna/pull/702/files#r350058631.

**Comment by hvy** [comment]
PTAL.

File: optuna/study.py (hunk 2 above)
Comment (hvy) [reply optuna/study.py]: Improved it slightly.
Screen Shot 2019-11-25 at 18 21 42

**Pull request merged**